server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java

```diff
@@ -93,7 +93,7 @@ private static NodeUsageStatsForThreadPools.ThreadPoolUsageStats replaceWritePoo
                 (float) writeLoadDelta,
                 writeThreadPoolStats.totalThreadPoolThreads()
             ),
-            writeThreadPoolStats.maxThreadPoolQueueLatencyMillis()
+            updateThreadPoolQueueLatencyWithShardMovements(writeThreadPoolStats.maxThreadPoolQueueLatencyMillis(), (float) writeLoadDelta)
```

Comment:
I am not sure this works. The simulator seems to be a new instance for every desired balance computation. So if we should have multiple such computations within a cluster info poll, this information is cleared for every one. The same goes for the simulated write load ofc.

I would think we want to tie this more to the cluster info cycle than the cluster-info-simulator? I could be missing something though.

Reply:
Yeah it is indeed currently reset in each `DesiredBalanceComputer#compute` invocation within a single `ClusterInfo` polling cycle. The same is true disk usages I think. I was assuming it was handled somewhere else, e.g. the [`getAdjustedDiskSpace`](https://github.com/elastic/elasticsearch/blob/5a6d3ae930dc3d5c7b9428e97d342ec9f006461f/server/src/main/java/org/elasticsearch/cluster/ClusterInfoSimulator.java#L42-L43) calls in `ClusterInfoSimulator`'s constructor. But upon closer look, I think they are for something else. So yes it indeed seems to be an issue and maybe addressed by one of the ideas proposed above, essentially retaining some information throughout the entire lifecycle of a single `ClusterInfo`. I think it should be a separate PR since it's a common issue.